test/tests/functional/pbs_wean_udp.py

```diff
@@ -0,0 +1,123 @@
+# coding: utf-8
+
+# Copyright (C) 2003-2020 Altair Engineering, Inc. All rights reserved.
+# Copyright notice does not imply publication.
+#
+# ALTAIR ENGINEERING INC. Proprietary and Confidential. Contains Trade Secret
+# Information. Not for use or disclosure outside of Licensee's organization.
+# The software and information contained herein may only be used internally and
+# is provided on a non-exclusive, non-transferable basis. License may not
+# sublicense, sell, lend, assign, rent, distribute, publicly display or
+# publicly perform the software or other information provided herein,
+# nor is Licensee permitted to decompile, reverse engineer, or
+# disassemble the software. Usage of the software and other information
+# provided by Altair(or its resellers) is only as explicitly stated in the
+# applicable end user license agreement between Altair and Licensee.
+# In the absence of such agreement, the Altair standard end user
+# license agreement terms shall govern.
+
+from tests.functional import *
+import socket
+
+class TestWeanUdp(TestFunctional):
+    """
+    Test suite consists of tests to check the functionality of pbs_comm daemon
+    """
+    def setUp(self):
+        TestFunctional.setUp(self)
+        #msg = "Need 2 mom hosts which are not present on server host"
+        #if len(self.moms) != 2 and self.server.hostname == self.mom.hostname:
+         #   self.skip_test(reason=msg)
+        self.pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        self.exec_path = os.path.join(self.pbs_conf['PBS_EXEC'], "bin")
+
+    def submit_job(self, set_attrib=None, exp_attrib=None, sleep=10, 
+                   interactive=False):
+        j = Job(PBSROOT_USER)
+        if set_attrib:
+           j.set_attributes(set_attrib)
+        if interactive:
+            j.interactive_script = [('hostname', '.*'),
+                                    ('export PATH=$PATH:%s' % self.exec_path, '.*'),
+                                     ('qstat', '.*')]
+        j.set_sleep_time(sleep)
+        jid = self.server.submit(j)
+        if exp_attrib:
+           self.server.expect(JOB, exp_attrib, id=jid)
+        return jid
+
+    def submit_resv(self, set_attrib=None, exp_attrib=None):
+        r = Reservation(PBSROOT_USER)
+        if set_attrib:
+           r.set_attributes(set_attrib)
+        rid = self.server.submit(r)
+        if exp_attrib:
+           self.server.expect(RESV, exp_attrib, id=rid)
+        return rid
+
+    def test_comm_with_mom(self):
+        """
+        Test the installation of communication daemon during PBS installation.
+        """
+        msg = "Need 2 moms as input. use -pmoms=<m1>:<m2>"
+        if len(self.moms) != 2:
+            self.skip_test(reason=msg)
+        (self.momA, self.momB) = self.moms.values()
```

Comment:
this will fail if -pmoms is more than 2

Reply:
but in that case, we would have skipped the test, just before we could execute this.